packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -650,6 +709,47 @@ export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
 
         return this.details;
     }
+
+    /**
+     * This should be called when running garbage collection to get details needed for
+     * idetifying which fluid objects to collect.
+     */
+    protected async getGarbageCollectioDetails(): Promise<IGarbageCollectionDetails> {
+        if (!this.garbageCollectionDetails) {
+            let tree: ISnapshotTree | null;
+            let garbageCollectionDetails: IGarbageCollectionDetails | undefined;
+
+            if (typeof this.initSnapshotValue === "string") {
```

Comment:
Couple thoughts: I think some of this logic is legacy and should not be copied over. And even if we need to have it, I'd rather refactor code to have it in one place, not duplicate it.

Reply:
Removed handling for older formats.